### PR TITLE
Analyze sheaf on p1

### PR DESCRIPTION
**Pull request by eisenbud**
This pull provides updates to CompleteIntersectionResolutions and also a new package, AnalyzeSheafOnP1 . The change log could say:

AnalyzeSheafOnP1 starts with a sheaf on P1 or a graded module over a polynomial ring in 2 variables, and computes the decomposition of the sheaf (or associated sheaf) into line bundles and cyclic skyscraper sheaves.

### CODE REVIEW

**Comment by DanGrayson** [comment]
It might be good to have a uniform word for this kind of thing.  We've been using ```factor``` for similar code:
```

     +---------------------------------------------------------------------------------------------+
     |i1 : f = random(ZZ^6, ZZ^4)                                                                  |
     |                                                                                             |
     |o1 = | 8 3 8 6 |                                                                             |
     |     | 1 7 5 8 |                                                                             |
     |     | 3 8 2 6 |                                                                             |
     |     | 7 8 3 9 |                                                                             |
     |     | 8 5 6 3 |                                                                             |
     |     | 3 7 3 7 |                                                                             |
     |                                                                                             |
     |              6        4                                                                     |
     |o1 : Matrix ZZ  <--- ZZ                                                                      |
     +---------------------------------------------------------------------------------------------+
     |i2 : M = subquotient ( f * diagonalMatrix{2,3,8,21}, f * diagonalMatrix{2*11,3*5*13,0,21*5} )|
     |                                                                                             |
     |o2 = subquotient (| 16 9  64 126 |, | 176 585  0 630 |)                                      |
     |                  | 2  21 40 168 |  | 22  1365 0 840 |                                       |
     |                  | 6  24 16 126 |  | 66  1560 0 630 |                                       |
     |                  | 14 24 24 189 |  | 154 1560 0 945 |                                       |
     |                  | 16 15 48 63  |  | 176 975  0 315 |                                       |
     |                  | 6  21 24 147 |  | 66  1365 0 735 |                                       |
     |                                                                                             |
     |                                 6                                                           |
     |o2 : ZZ-module, subquotient of ZZ                                                            |
     +---------------------------------------------------------------------------------------------+
     |i3 : factor M                                                                                |
     |                                                                                             |
     |          ZZ   ZZ    ZZ                                                                      |
     |o3 = ZZ + -- + -- + ----                                                                     |
     |           5   11   5*13                                                                     |
     |                                                                                             |
     |o3 : Expression of class Sum                                                                 |
     +---------------------------------------------------------------------------------------------+
```

**Comment by eisenbud** [comment]
It occurred to me that "decomposeSheafOnP1" would be better than what I did; I'd be happy to change to that. "Factor" is a bit farther afield, it seems to me. You already have some decomposes in your system too, I think.
--
David Eisenbud
Director, Mathematical Sciences Research Institute; and
Professor of Mathematics,University of California, Berkeley
www.msri.org/~de

**Comment by DanGrayson** [comment]
"decompose" might make sense, too, but "decomposeSheafOnP1" is too wordy.

When someone else writes a package for decomposing sheaves on P2 we'll have
to
have a common intermediary, but until then, why not?

**Comment by zteitler** [comment]
How about something like "GrothendieckDecomposition"? (Well, it breaks the first-word-lower-case convention and it doesn't incorporate the analysis of skyscraper sheaves, so perhaps never mind.)

**Comment by eisenbud** [comment]
I don't think that package name is too wordy -- we need descriptiveness! The actual routine within it is currently called "analyze". Perhaps "decompose" is already taken by some form of primary decomposition?

David
--
David Eisenbud
Director, Mathematical Sciences Research Institute; and
Professor of Mathematics,University of California, Berkeley
www.msri.org/~de

**Comment by mikestillman** [comment]
I agree, I think that wordiness is better here.  In any case, decomposing sheaves on P^1 is a special (useful) function, which won’t generalize (if I understand what it is doing), so having a name special to that is fine, it seems.

**Comment by DanGrayson** [comment]
I've been talking about the routine name all along, not the package name,
which should be descriptive, as you say.  I guess I should have looked at
the source code, but instead I took my clue as to its name from your
description: "AnalyzeSheafOnP1 starts with a sheaf on P1 or a graded module
over a polynomial ring in 2 variables, and ...".  Sorry!

"DecomposeSheafOnP1" and "decompose" might be better names, in light of the
definition of indecomposable module.  Currently, the documentation for
"decompose" says it's a synonym for "minimalPrimes", but that
is no longer true, and it would be an easy matter to change the
documentation for "decompose" to be
more neutral about this umbrella word.

**Comment by DanGrayson** [comment]
You have inadvertently checked in a file  M2/Macaulay2/m2/AnalyzeSheafOnP1.m2  and also checked in a change to the commit number of the submodule mathic.

**Comment by mikestillman** [comment]
Maybe it was me that screwed up the mathic update, yesterday I had a conflict that I didn't know how to fix, but at one point the conflict went away...

**Comment by DanGrayson** [comment]
I've hand selected the appropriate parts and committed them here: https://github.com/Macaulay2/M2/commit/4866b1073125937e1e82ce533eb36395724c3aec

Please check I did the right thing.
